packages/xstate-graph/src/types.ts

```diff
@@ -57,72 +56,112 @@ export type DirectedGraphNode = JSONSerializable<
   }
 >;
 
-export interface AdjacencyMap<TContext, TEvent extends EventObject> {
-  [stateId: string]: Record<
-    string,
+export interface AdjacencyMap<TState, TEvent extends EventObject> {
+  [stateId: SerializedState]: Record<
+    SerializedState,
     {
-      state: State<TContext, TEvent>;
+      state: TState;
       event: TEvent;
     }
   >;
 }
 
-export interface StatePaths<TContext, TEvent extends EventObject> {
+export interface StatePlan<TState, TEvent extends EventObject> {
   /**
    * The target state.
    */
-  state: State<TContext, TEvent>;
+  state: TState;
   /**
    * The paths that reach the target state.
    */
-  paths: Array<StatePath<TContext, TEvent>>;
+  paths: Array<StatePath<TState, TEvent>>;
 }
 
-export interface StatePath<TContext, TEvent extends EventObject> {
+export interface StatePath<TState, TEvent extends EventObject> {
   /**
    * The ending state of the path.
    */
-  state: State<TContext, TEvent>;
+  state: TState;
   /**
-   * The ordered array of state-event pairs (segments) which reach the ending `state`.
+   * The ordered array of state-event pairs (steps) which reach the ending `state`.
    */
-  segments: Segments<TContext, TEvent>;
+  steps: Steps<TState, TEvent>;
   /**
-   * The combined weight of all segments in the path.
+   * The combined weight of all steps in the path.
    */
   weight: number;
 }
 
-export interface StatePathsMap<TContext, TEvent extends EventObject> {
-  [key: string]: StatePaths<TContext, TEvent>;
+export interface StatePlanMap<TState, TEvent extends EventObject> {
+  [key: string]: StatePlan<TState, TEvent>;
 }
-export interface Segment<TContext, TEvent extends EventObject> {
+
+export interface Step<TState, TEvent extends EventObject> {
   /**
    * The current state before taking the event.
    */
-  state: State<TContext, TEvent>;
+  state: TState;
   /**
    * The event to be taken from the specified state.
    */
   event: TEvent;
 }
 
-export type Segments<TContext, TEvent extends EventObject> = Array<
-  Segment<TContext, TEvent>
+export type Steps<TState, TEvent extends EventObject> = Array<
+  Step<TState, TEvent>
 >;
 
 export type ExtractEvent<
   TEvent extends EventObject,
   TType extends TEvent['type']
 > = TEvent extends { type: TType } ? TEvent : never;
 
-export interface ValueAdjMapOptions<TContext, TEvent extends EventObject> {
+export interface ValueAdjMapOptions<TState, TEvent extends EventObject> {
```

Comment:
q: why is this a `ValueAdjMapOptions` and not `AdjMapOptions`? 🤔 

as a sidenote - it would be cool to not use the `adj` abbreviation in any type name - I remember that when I was first reading the codebase I had to look up what the adj stands for and I couldn't simply google it because it was a custom~ abbreviation, whereas even if I didn't know what adjacency is i would be able to google for it

Reply:
Good point regarding `adj` - I'll spell it out.

Reply:
Added as a task in the description